specification/synapse/data-plane/Microsoft.Synapse/preview/2019-11-01-preview/adf/entityTypes/NoteBook.json

```diff
@@ -0,0 +1,251 @@
+      "description": "Notebook cell.",
+      "type": "object",
+      "properties": {
+        "cell_type": {
+          "description": "String identifying the type of cell.",
+          "type": "string"
+        },
+        "metadata": {
+          "description": "Cell-level metadata.",
+          "type": "object"
+        },
+        "source": {
+          "description": "Contents of the cell, represented as an array of lines.",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "attachments": {
+          "description": "Attachments associated with the cell.",
+          "type": "object"
+        },
+        "outputs": {
+          "description": "Cell-level output items.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/NotebookCellOutputItem"
+          }
+        }
+      },
+      "additionalProperties": {
+        "type": "object"
+      },
+      "required": [
+        "cell_type",
+        "metadata",
+        "source"
+      ]
+    },
+    "NotebookSessionProperties": {
+      "description": "Session properties.",
+      "properties": {
+        "driverMemory": {
+          "description": "Amount of memory to use for the driver process.",
+          "type": "string"
+        },
+        "driverCores": {
+          "description": "Number of cores to use for the driver.",
+          "type": "integer"
+        },
+        "executorMemory": {
+          "description": "Amount of memory to use per executor process.",
+          "type": "string"
+        },
+        "executorCores": {
+          "description": "Number of cores to use for each executor.",
+          "type": "integer"
+        },
+        "numExecutors": {
+          "description": "Number of executors to launch for this session.",
+          "type": "integer"
+        }
+      },
+      "required": [
+        "driverMemory",
+        "driverCores",
+        "executorMemory",
+        "executorCores",
+        "numExecutors"
+      ]
+    },
+    "NotebookCellOutputItem": {
+      "description": "An item of the notebook cell execution output.",
+      "properties": {
+        "name": {
+          "description": "For output_type=stream, determines the name of stream (stdout / stderr).",
+          "type": "string"
+        },
+        "execution_count": {
+          "description": "Execution sequence number.",
+          "type": "integer"
+        },
+        "output_type": {
+          "description": "Execution, display, or stream outputs.",
+          "type": "string",
+          "enum": [
+            "execute_result",
+            "display_data",
+            "stream",
+            "error"
+          ],
+          "x-ms-enum": {
+            "name": "CellOutputType",
+            "modelAsString": true
+          }
+        },
+        "text": {
+          "description": "For output_type=stream, the stream's text output, represented as a string or an array of strings."
```

Comment:
I got a warning from autorest here, that a type should be specified:
- If the actual JSON content is really union(string, array[string]), then put "object"
- If this is always a string, then put "string"

Reply:
> I got a warning from autorest here, that a type should be specified:
> 
> * If the actual JSON content is really union(string, array[string]), then put "object"
> * If this is always a string, then put "string"

Fixed enum type to string. The other type will double check with RP team.